gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -153,9 +153,13 @@ private static void configureSourceSet(
         });
 
         explicitCompile.withDependencies(deps -> {
-            // Mirror the transitive constraints form compileClasspath in order to pick up GCV locks.
-            // We should really do this with an addAllLater but that would require Gradle 6, or a hacky workaround.
-            explicitCompile.getDependencyConstraints().addAll(compileClasspath.getAllDependencyConstraints());
+            // Pick up GCV locks. We're making an internal assumption that this configuration exists,
+            // but we can rely on this since we control GCV.
+            // Alternatively, we could tell GCV to lock this configuration, at the cost of a slightly more
+            // expensive 'unifiedClasspath' resolution during lock computation.
+            if (project.getRootProject().getPluginManager().hasPlugin("com.palantir.versions-lock")) {
+                explicitCompile.extendsFrom(project.getConfigurations().getByName("lockConstraints"));
```

Comment:
Feels quite weird to have this direct knowledge of GCV here? Will this still work with old GCV?